testcases/OpTestGSBStaticKey.py

```diff
@@ -0,0 +1,299 @@
+        # Variables required for this test
+        self.os_secureboot = False
+        self.dt_secureboot = False
+        self.lockdown = False
+        self.sys_lockdown = ""
+        self.grubFilename = ""
+        self.prepDisk = ""
+        self.signature = "~Module signature appended~"
+
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        os_level = self.cv_HOST.host_get_OS_Level()
+        if 'Red Hat' in os_level:
+            self.distro = 'rhel'
+        elif 'SLES' in os_level:
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported on "
+                                "SLES and RHEL releases")
+
+    def check_hmc_secureboot_state(self):
+        '''
+        Return
+        'True' in case of Secure boot enabled
+        'False' in case of Secure boot disabled
+        '''
+        # HMC command to know the current state of Secure Boot
+        cmd = ("lssyscfg -r lpar -m %s -F curr_secure_boot --filter "
+               "lpar_names=%s" %
+               (self.cv_HMC.mg_system, self.cv_HMC.lpar_name))
+        output = self.hmc_con.run_command(cmd, timeout=300)
+        if int(output[0]) == 2: # Value '2' means Secure Boot enabled
+            return True
+        elif int(output[0]) == 0: # Value '0' means Secure Boot disabled
+            return False
+
+    def hmc_secureboot_on_off(self, enable=True):
+        '''
+        Enable/Disable Secure Boot from HMC
+        1. PowerOFF/Shutdown LPAR from HMC
+        2. Enable/Disable Secure boot using 'chsyscfg' command
+        3. PowerON/Activate the LPAR and boot to Operating System
+        '''
+        # PowerOFF/shutdown LPAR from HMC
+        self.cv_HMC.poweroff_lpar()
+        # Set Secure Boot value using HMC command
+        cmd = ('chsyscfg -r lpar -m %s -i "name=%s, secure_boot=' %
+               (self.cv_HMC.mg_system, self.cv_HMC.lpar_name))
+        if enable: # Value '2' to enable Secure Boot
+            cmd = '%s2"' % cmd
+        else: # Value '0' to disable Secure Boot
+            cmd = '%s0"' % cmd
+        # Execute the command
+        self.hmc_con.run_command(cmd, timeout=300)
+        # PowerON/Activate LPAR and Boot the Operating System
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        # Checking What Secure Boot state got set,
+        # according to the required enable/disable
+        hmc_secureboot = self.check_hmc_secureboot_state()
+        if enable and not hmc_secureboot:
+                self.fail("HMC: Failed to enable Secure Boot")
+        elif not enable and hmc_secureboot:
+                self.fail("HMC: Failed to disable Secure Boot")
+
+    def getPRePDisk(self):
+        '''
+        Identify the PReP partition, this disk name is required to copy
+        signed grub. This manual step required for RHEL OS only.
+        '''
+        cmd = 'sfdisk -l | grep "PPC PReP Boot" | tail -1 | cut -d" " -f1'
```

Comment:
can we explore python's way as running bash and using pipe statement is resource intensive

Reply:
Sure.